test/rpc/test_rpc_misc.py

```diff
@@ -121,3 +150,113 @@ def test_call_with_invalid_params(params):
+            },
+            4,
+        ),
+        (
+            [*PREDEPLOY_ACCOUNT_CLI_ARGS],
+            {
+                "from_block": "0",
+                "to_block": "3",
+            },
+            2,
+        ),
+        (
+            [*PREDEPLOY_ACCOUNT_CLI_ARGS],
+            {
+                "from_block": "3",
+                "to_block": "4",
+            },
+            2,
+        ),
+        (
+            [*PREDEPLOY_ACCOUNT_CLI_ARGS],
+            {
+                "from_block": "0",
+                "to_block": "latest",
+                "chunk_size": 3,
+            },
+            3,
+        ),
+        (
+            [*PREDEPLOY_ACCOUNT_CLI_ARGS],
+            {
+                "from_block": "0",
+                "to_block": "latest",
+                "chunk_size": 3,
+                "continuation_token": "1",
+            },
+            1,
+        ),
+        (
+            [*PREDEPLOY_ACCOUNT_CLI_ARGS],
+            {
+                "from_block": "0",
+                "to_block": "latest",
+                "address": "0x62230eA046a9a5fbc261ac77d03c8d41e5d442db2284587570ab46455fd2488",
+            },
+            2,
+        ),
+        (
+            [*PREDEPLOY_ACCOUNT_CLI_ARGS],
+            {
+                "from_block": "0",
+                "to_block": "latest",
+                "keys": [
+                    "0x99cd8bde557814842a3121e8ddfd433a539b8c9f14bf31ebf108d12e6196e9"
+                ],
+            },
+            2,
+        ),
+        (
+            [*PREDEPLOY_ACCOUNT_CLI_ARGS],
+            {
+                "from_block": "0",
+                "to_block": "latest",
+                "keys": [
+                    "0x99cd8bde557814842a3121e8ddfd433a539b8c9f14bf31ebf108d12e6196e9",
+                    "0x3db3da4221c078e78bd987e54e1cc24570d89a7002cefa33e548d6c72c73f9d",
+                ],
+            },
+            4,
+        ),
+        (
+            [*PREDEPLOY_ACCOUNT_CLI_ARGS],
+            {
+                "from_block": "0",
+                "to_block": "latest",
+                "address": "0x62230eA046a9a5fbc261ac77d03c8d41e5d442db2284587570ab46455fd2488",
+                "keys": [
+                    "0x99cd8bde557814842a3121e8ddfd433a539b8c9f14bf31ebf108d12e6196e9",
+                    "0x3db3da4221c078e78bd987e54e1cc24570d89a7002cefa33e548d6c72c73f9d",
+                ],
+            },
+            2,
+        ),
+    ],
+    indirect=True,
+)
+def test_get_events(input_data, expected_data):
+    """
+    Test RPC get_events.
+    """
+    deploy_info = deploy(EVENTS_CONTRACT_PATH)
+    for i in range(0, 2):
+        invoke(
+            calls=[(deploy_info["address"], "increase_balance", [i])],
+            account_address=PREDEPLOYED_ACCOUNT_ADDRESS,
+            private_key=PREDEPLOYED_ACCOUNT_PRIVATE_KEY,
+        )
+    resp = rpc_call("starknet_getEvents", params=input_data)
+    assert len(resp["result"]["events"]) == expected_data
```

Comment:
This ensures that the number of events returned is the number of events expected.
As we are not testing events but filtering I agree that we don't need to check Event structure, but also that the same filtering can return the right number of events but wrong ones (miss order, broken mapping, new starknet version break)?
Could we check that we get exactly the events we expect to get for each case?

Reply:
I was thinking about that also, one time I have a false positive like this when I was implementing chunking.

I'll try to figure out something better.

Reply:
Changed for checking if events data are equal